packages/auth/plugin/src/ios/urlTypes.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  ConfigPlugin,
+  IOSConfig,
+  withInfoPlist,
+  ExportedConfigWithProps,
+} from '@expo/config-plugins';
+import fs from 'fs';
+import path from 'path';
+import plist from 'plist';
+
+// does this for you: https://firebase.google.com/docs/auth/ios/phone-auth#enable-phone-number-sign-in-for-your-firebase-project
+export const withIosCaptchaUrlTypes: ConfigPlugin = config => {
+  return withInfoPlist(config, config => {
+    if (!config.ios?.googleServicesFile) {
+      throw new Error(
+        'Path to GoogleService-Info.plist is not defined. Please specify the `expo.ios.googleServicesFile` field in app.json.',
+      );
+    }
+
+    return setUrlTypesForCaptcha({ config });
+  });
+};
+
+export function setUrlTypesForCaptcha({
```

Comment:
I'd add unit tests for this method. To keep it more simple, I'd split it into two functions:
- `getReservedClientId(googleServicesFilePath: string): string`
- `addUriScheme(config, scheme): config`
Then provide fixtures for `GoogleServices-Info.plist` and some simple config object `const config = { ios: { infoPlist: { ... } }` and compare outputs.